app/uploaders/article_image_uploader.rb

```diff
@@ -0,0 +1,53 @@
+class ArticleImageUploader < CarrierWave::Uploader::Base
+  include CarrierWave::MiniMagick
+
+  def store_dir
+    # This will be the directory path in the CMS
+    "uploads/#{model.class.to_s.underscore}/#{mounted_as}/#{model.id}"
+  end
+
+  def extension_whitelist
+    %w[jpg jpeg gif png]
+  end
+
+  def filename
+    # don't try changing the name if original_filename is missing
+    return unless original_filename
+
+    # This will be the filename for all the processed/uploaded images
+    model.title.downcase.tr(' ', '_') + File.extname(original_filename)
```

Comment:
We usually name our headers `header.jpg`. 

Though if there are two articles on the same day, we name them `header.jpg` and `header-2.jpg`.

Could this Do The Right Thing™ and check to see if there are other articles on that same day and adjust the name incrementer accordingly? Or is that a hard thing?